contract-testing/tests/contract_queries.rs

```diff
@@ -0,0 +1,363 @@
+//! This module contains tests for smart contract module reference and name
+//! queries. These tests match equivalent tests for the scheduler in the node.
+use std::io::Write;
+
+use concordium_smart_contract_testing::*;
+mod helpers;
+
+#[macro_export]
+/// Macro for a pattern that matches a [ContractInvokeError] where the contract
+/// rejects with the supplied error code.
+macro_rules! ContractInvokeErrorCode {
+    ($x:expr) => {
+        ContractInvokeError {
+            kind: ContractInvokeErrorKind::ExecutionError {
+                failure_kind: InvokeFailure::ContractReject {
+                    code: $x,
+                    ..
+                },
+            },
+            ..
+        }
+    };
+}
```

Comment:
It would be better to add a general helper method to `ContractInvokeError` since this is

```rust
/// If invocation failed with an execution error due to contract logic return the return status code and return value.
/// The return status will always be negative.
pub fn logic_reject(&self) -> Option<(i32, &[u8])>) {
   ..
}
```

These kind of helpers are useful.

And then just
```rust
matches!(res.logic_reject(), Some((-1, _)))
```